client/src/main/java/com/netflix/conductor/client/automator/TaskPollExecutor.java

```diff
@@ -357,4 +398,31 @@ private PollingSemaphore getPollingSemaphore(String taskType) {
             return pollingSemaphoreMap.get(ALL_WORKERS);
         }
     }
+
+    private Runnable extendLease(Task task, CompletableFuture<Task> taskCompletableFuture) {
+        return () -> {
+            if (taskCompletableFuture.isDone()) {
+                LOGGER.warn(
+                        "Task processing for {} completed, but its lease extend was not cancelled",
+                        task.getTaskId());
+                return;
+            }
+            LOGGER.info("Attempting to extend lease for {}", task.getTaskId());
+            try {
+                TaskResult result = new TaskResult(task);
+                result.getLogs().add(new TaskExecLog("lease extend"));
+                result.setExtendLease(true);
+                retryOperation(
+                        (TaskResult taskResult) -> {
+                            taskClient.updateTask(taskResult);
+                            return null;
+                        },
+                        LEASE_EXTEND_RETRY_COUNT,
+                        result,
+                        "extend lease");
```

Comment:
It would be useful to add a metric here to capture lease extensions based on task type.